src/components/NeedFundsNotice.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { fromWei } from 'web3-utils';
+
+import Highlighted from './Highlighted';
+import CopiableAddress from './CopiableAddress';
+import WarningBox from './WarningBox';
+import Blinky from './Blinky';
+
+export default function NeedFundsNotice({
+  address,
+  minBalance,
+  balance,
+  ...rest
+}) {
+  return (
+    <WarningBox {...rest}>
+      <Highlighted warning>
+        Your ownership address <CopiableAddress>{address}</CopiableAddress>{' '}
+        needs at least {fromWei(minBalance)} ETH and currently has{' '}
+        {fromWei(balance)} ETH. The transaction will automatically resume once
+        ETH is available. Waiting... <Blinky />
```

Comment:
Is this maybe missing the trailing `{' '}`, or is that only in/around interpolation of any kind? The sentence reads a bit poorly anyway, should probably say "once _enough_ ETH is available".

Reply:
the trailing `{' '}` is only necessary when, by default, react would collapse the newline into no space at all. by default a newline in jsx between non-interpolated values will be treated as a normal space. so yeah, pretty much. my editor automatically adds these where necessary, though, so I'm trusting it a little bit.